### PR TITLE
chore: turn on no-shadow lint rule

### PR DESCRIPTION
**Pull request by kumar303**
One of our contributors got tripped up on overshadowed variables (https://github.com/mozilla/web-ext/pull/552#discussion_r82895201) and Flow might even have some bugs around it.

### CODE REVIEW

**Comment by kumar303** [comment]
@rpl what do you think? Hopefully not too annoying?

**Comment by kumar303** [comment]
@rpl will you have some time soon to take a look? No rush.

File: tests/unit/test-firefox/test.firefox.js [hunk 1]
```diff
@@ -38,14 +38,11 @@ describe('firefox', () => {
     };
 
     function createFakeFxRunner(firefoxOverrides = {}) {
-      const firefox = {
+      const process = {
```
Comment (rpl) [inline tests/unit/test-firefox/test.firefox.js]: How about calling it `firefoxProcess` or `fxProcess` instead? 
`process` is going to shadow the `process` identifier as defined by nodejs 
(that's a pity that the eslint `no-shadow` rule doesn't seem to have any config available for customized disallowed identifier)

**Comment by kumar303** [comment]
Good call on the built-ins, thanks. That exposed a few more problems. I think https://github.com/mozilla/web-ext/pull/744/ is close to merge so I'm going to wait on that before landing this one just in case more fixes are needed.

**Pull request merged**